Rest top20 sellers

demotivirus opened the pull request:
[{"userName":"vasya.pupkin","countOfSell":null},{"userName":"petya.vakulov","countOfSell":null},{"userName":"semen.semenych","countOfSell":null},{"userName":"mikailo.lomonosov","countOfSell":null}]

demotivirus committed:
Создал рест
Вывод типа:
[{"userName":"vasya.pupkin","countOfSell":null},{"userName":"petya.vakulov","countOfSell":null},{"userName":"semen.semenych","countOfSell":null},{"userName":"mikailo.lomonosov","countOfSell":null}]

demotivirus committed:
Merge branch 'dev' into rest-top20-sellers

NikitaNesterenko commented on dao-impl/src/main/java/jm/stockx/api/dao/SellingDAOImpl.java:
```diff
@@ -1,8 +1,19 @@
 @Repository
 public class SellingDAOImpl extends AbstractDAO<SellingInfo, Long> implements SellingDAO {
+    @Override
+    public List<User> getTop20SellingUsers(){
```
переименовать метод, без 20

NikitaNesterenko commented on service-impl/src/main/java/jm/stockx/SellingInfoServiceImpl.java:
```diff
@@ -0,0 +1,28 @@
+    public List<SellerTopInfoDto> getTop20SellingUsers() {
+        return sellingDAO.getTop20SellingUsers()
+                .stream()
+                .map(m -> new SellerTopInfoDto(m.getUsername()))
```
сделать все через Tuple в дао, без доставания всех юзеров, тащи ток необходимые поля

demotivirus committed:
Переписал как смог
[{"userName":"vasya.pupkin","countOfSell":4},{"userName":"petya.vakulov","countOfSell":3},{"userName":"semen.semenych","countOfSell":1},{"userName":"mikailo.lomonosov","countOfSell":1}]
Буду копать дальше

demotivirus committed:
Merge branch 'dev' into rest-top20-sellers

demotivirus committed:
Переписал как смог
[{"userName":"vasya.pupkin","countOfSell":4},{"userName":"petya.vakulov","countOfSell":3},{"userName":"semen.semenych","countOfSell":1},{"userName":"mikailo.lomonosov","countOfSell":1}]
Буду копать дальше

NikitaNesterenko commented on dao-impl/src/main/java/jm/stockx/api/dao/SellingDAOImpl.java:
```diff
@@ -1,8 +1,29 @@
+    @Override
+    public List<User> getTopSellingUsers(){
+        String sql = "select i.* from selling_info as si left join users as i on si.user_id=i.id " +
+                " group by si.user_id order by count(si.item_id) desc";
```
формировать ДТО прям в запросе, можно написать на JPQL через SELECT new DTO...
